### PR TITLE
chore: move from inet.af/netaddr to net/netip and go4.org/netipx

### LINKED ISSUE #6007: netaddr -> netip refactoring
Refactor (if possible) Talos to use our own fork of `netaddr` which is compatible with `netip` package.

### PR DESCRIPTION
**Pull request by DmitriyMV**
Closes #6007

Signed-off-by: Dmitriy Matrenichev <dmitry.matrenichev@siderolabs.com>

### CODE REVIEW

**Comment by DmitriyMV** [comment]
There is only one instance of `netaddr` remaining - in [internal/app/machined/pkg/controllers/runtime/kmsg_log_test.go:24](https://github.com/siderolabs/talos/blob/master/internal/app/machined/pkg/controllers/runtime/kmsg_log_test.go#L24). That's because siderolink has talos in deps, so we need to update talos first.

File: go.mod [hunk 1]
```diff
@@ -269,6 +269,7 @@ require (
 	go.starlark.net v0.0.0-20200306205701-8dd3e2ee1dd5 // indirect
 	go.uber.org/multierr v1.8.0 // indirect
 	go4.org/intern v0.0.0-20220617035311-6925f38cc365 // indirect
+	go4.org/netipx v0.0.0-20220812043211-3cc044ffd68d // indirect
```
Comment (smira) [inline go.mod:272]: should it be direct dependency?

File: pkg/machinery/nethelpers/std_netaddr.go [hunk 2]
```diff
@@ -6,24 +6,25 @@ package nethelpers
 
 import (
 	"net"
+	"net/netip"
 
-	"inet.af/netaddr"
```
Comment (smira) [inline pkg/machinery/nethelpers/std_netaddr.go:10]: nit: this file might be renamed to `std_netip.go`

**Review by smira — APPROVED** [review]
also due to massive changes, it might make sense to `/promote integration`

File: go.mod (hunk 1 above)
Comment (DmitriyMV) [reply go.mod]: Rerun `go mod tidy` - should be good now.

File: pkg/machinery/nethelpers/std_netaddr.go (hunk 2 above)
Comment (DmitriyMV) [reply pkg/machinery/nethelpers/std_netaddr.go:10]: Done!

**Pull request merged**